pkg/controller/controller_test.go

```diff
@@ -1229,6 +1382,31 @@ func TestProvision(t *testing.T) {
 			expectErr:         true,
 			expectState:       controller.ProvisioningNoChange,
 		},
+		"fail to get secret reference for default secret parameter template with annotation": {
```

Comment:
Is the failure caused by mismatch between `secret-from-annotation` and `default-secret-from-annotation`? If so, could you make the names more distinct? Thanks.

Reply:
As @msau42 pointed out the default secrets can support only that set of template variables which is common for all calls. In this test, I showed that the annotation variable is not supported by the default secret. Should I rename this test?

Reply:
I think after that explanation, the test name makes sense. I just find the existing logic with secret references and template evaluation to be confusing. :cry:

Reply:
Now that I re-read the test, maybe we can update the name to mention invalid/illegal template variable.

Reply:
Fixed.